src/Donations/Components/DonationsForm.tsx

```diff
@@ -187,6 +196,58 @@ function DonationsForm() {
     }
   }
 
+  const handlePlanetCashDonate = async () => {
+    const _onBehalfDonor = {
+      firstname: onBehalfDonor.firstName,
+      lastname: onBehalfDonor.lastName,
+      email: onBehalfDonor.email,
+    };
+
+    // create Donation data
+    const donationData = {
+      purpose: projectDetails!.purpose,
+      project: projectDetails!.id,
+      units: quantity,
+      prePaid: true,
+      onBehalf: onBehalf,
+      ...(onBehalf && { donor: _onBehalfDonor }),
+      ...(isGift && { gift: giftDetails }),
+    };
+
+    const cleanedDonationData = cleanObject(donationData);
+
+    const token = await getAccessTokenSilently();
+
+    // @method    POST
+    // @endpoint  /app/donation
+
+    try {
+      await apiRequest({
+        url: "/app/donations",
+        method: "POST",
+        setshowErrorCard,
+        data: cleanedDonationData,
+        token,
+        headers: {
+          "IDEMPOTENCY-KEY": uuidv4(),
+        },
+      });
+    } catch (err) {
+      console.error(err);
+      if (err.status === 400) {
+        setPaymentError(err.data.message);
+      } else if (err.status === 500) {
+        setPaymentError("Something went wrong please try again soon!");
```

Comment:
Can you add translation resources to these messages? Someone from the team will translate it to other languages.

Reply:
Done